src/EditorFeatures/DiagnosticsTestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -122,7 +122,14 @@ public void TestSupportedDiagnosticsMessageHelpLinkUri()
 
                 foreach (var descriptor in diagnosticAnalyzer.SupportedDiagnostics)
                 {
-                    Assert.NotEqual("", descriptor.HelpLinkUri ?? "");
+                    if (descriptor.Id is "IDE0043" or "IDE1007" or "IDE1008" or "RemoveUnnecessaryImportsFixable" or "RE0001")
```

Comment:
📝 Rather than exclude `RemoveUnnecessaryImportsFixable` directly, we should skip analysis of descriptors meeting both of:

* Severity set to `DiagnosticSeverity.Hidden`
* Tags contains `WellKnownDiagnosticTags.NotConfigurable`

These diagnostics are never visible in the UI.

Reply:
@sharwell You mean in the following?

https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDConfigurationTests.cs#L75-L81

- This is a bit out-of-scope to do in #51859
- This will skip IDE0005, which indeed has a help url

Reply:
No, I'm referring specifically to `TestSupportedDiagnosticsMessageHelpLinkUri`.

Reply:
@sharwell The change done here was just to show what other failures remain before deleting the test in #51859 per your request in https://github.com/dotnet/roslyn/pull/51859#discussion_r594270292.

This PR isn't supposed to get merged.

Reply:
I think I'd prefer to merge this one rather than delete the test. It should be possible by ignoring descriptors with the above property and updating this descriptor to include a help link that just points back to <https://github.com/dotnet/roslyn>:

https://github.com/dotnet/roslyn/blob/73e98581cc6034e93812dc37e7cde7aa93e11f47/src/EditorFeatures/CSharpTest/Diagnostics/MockDiagnosticAnalyzerTests.cs#L24

Reply:
Does the test add any value given that it duplicates the work done in https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDConfigurationTests.cs#L42 (which is even more general as it handles both new and old tests).

Reply:
My main concern is that test only covers a subset of potential analyzer assemblies, and centers around a different primary goal. I wouldn't go so far as to _remove_ the other test, but I'm not sure I'd call it a duplicate in its current form.

Reply:
> My main concern is that test only covers a subset of potential analyzer assemblies

Makes sense. But that would mean *both* tests are not good enough, as `TestSupportedDiagnosticsMessageHelpLinkUri` don't run for the new testing library, which we have a goal to move most/all tests to it.

So probably this should be done in few steps:

- Un-skip the test for now.
- See how to improve `ValidateHelpLinkForDiagnostic` (or find a replacement to it) that would cover all assemblies.
- Delete `TestSupportedDiagnosticMessageHelpLinkUri` in favor of what's done in the previous step.

What do you think?

Reply:
Yes, though I'm thinking the last step would wait until we're ready to delete `AbstractDiagnosticProviderBasedUserDiagnosticTest` (because all tests have moved to the new framework).